### PR TITLE
Loop Index.js scripts to find token

### PR DESCRIPTION
**Pull request by Jacrispys**
Fix for issue #37 

Note: I haven't been able to compile and test this fix since the 
`com.github.freyacodes.lavalink:Lavalink-Server:3.6.0`
implementation in the plugin gradle doesn't exist for me ¯\_(ツ)_/¯

### CODE REVIEW

**Review by TopiSenpai — CHANGES_REQUESTED** [review]
You should look through every index and try get the token. As for the src attribute it's on every element returned as we specifically filter for that.


As for testing you can compile the plugin & throw it into the plugins directory. Lavalink is annoyingly still broken on jitpack

**Comment by Jacrispys** [comment]
Ready for review, built project and IMPL worked.

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java [hunk 1]
```diff
@@ -123,28 +124,35 @@ public void parseTokenData() throws IOException {
 
     public void requestToken() throws IOException {
         var request = new HttpGet("https://music.apple.com");
-        String tokenScriptURL;
+        String tokenScriptURL = null;
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java:127]: this has no use here

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java [hunk 2]
```diff
@@ -123,28 +124,35 @@ public void parseTokenData() throws IOException {
...
         try (var response = this.httpInterfaceManager.getInterface().execute(request)) {
             var document = Jsoup.parse(response.getEntity().getContent(), null, "");
-            var element = document.selectFirst("script[type=module][src~=/assets/index.*.js]");
-            if (element == null) {
+            var elements = document.select("script[type=module][src~=/assets/index.*.js]");
+            if (elements.size() == 0) {
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java:131]:
```suggestion
            if (elements.isEmpty()) {
```

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java [hunk 3]
```diff
@@ -123,28 +124,35 @@ public void parseTokenData() throws IOException {
...
-        try (var response = this.httpInterfaceManager.getInterface().execute(request)) {
-            var tokenScript = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8);
-            var tokenMatcher = TOKEN_SCRIPT_PATTERN.matcher(tokenScript);
-            if (!tokenMatcher.find()) {
-                throw new IllegalStateException("Cannot find token in script");
+            List<Element> elementList = new ArrayList<>(elements);
+
+            boolean foundToken = false;
+
+            for (Element element : elementList) {
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java:138]: use var here

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java [hunk 4]
```diff
@@ -123,28 +124,35 @@ public void parseTokenData() throws IOException {
...
+                tokenScriptURL = element.attr("src");
+                if (!tokenScriptURL.isEmpty()) {
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java:140]: this should never be empty as we query specifically for the `src` attribute

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java [hunk 5]
```diff
@@ -123,28 +124,35 @@ public void parseTokenData() throws IOException {
...
+                    request = new HttpGet("https://music.apple.com" + tokenScriptURL);
+                    try (var indexResponse = this.httpInterfaceManager.getInterface().execute(request)) {
+                        var tokenScript = IOUtils.toString(indexResponse.getEntity().getContent(), StandardCharsets.UTF_8);
+                        var tokenMatcher = TOKEN_SCRIPT_PATTERN.matcher(tokenScript);
+                        if (tokenMatcher.find()) {
+                            foundToken = true;
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java:146]: if you find the token just return. No need for `foundToken`

**Review by TopiSenpai — CHANGES_REQUESTED** [review]

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java [hunk 6]
```diff
@@ -123,28 +124,35 @@ public void parseTokenData() throws IOException {
...
-        if (tokenScriptURL.isEmpty()) {
-            throw new IllegalStateException("Cannot find token script url");
-        }
-        request = new HttpGet("https://music.apple.com" + tokenScriptURL);
-        try (var response = this.httpInterfaceManager.getInterface().execute(request)) {
-            var tokenScript = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8);
-            var tokenMatcher = TOKEN_SCRIPT_PATTERN.matcher(tokenScript);
-            if (!tokenMatcher.find()) {
-                throw new IllegalStateException("Cannot find token in script");
+            List<Element> elementList = new ArrayList<>(elements);
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java:134]: why copy the list here?

Comment (Jacrispys) [reply main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java]: oh I didn't realize  that I could use  `Elements` as a foreach for `Element` using vars I suppose I can though.. Changing now.

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java [hunk 7]
```diff
@@ -124,34 +119,29 @@ public void parseTokenData() throws IOException {
...
-                tokenScriptURL = element.attr("src");
-                if (!tokenScriptURL.isEmpty()) {
-                    request = new HttpGet("https://music.apple.com" + tokenScriptURL);
-                    try (var indexResponse = this.httpInterfaceManager.getInterface().execute(request)) {
-                        var tokenScript = IOUtils.toString(indexResponse.getEntity().getContent(), StandardCharsets.UTF_8);
-                        var tokenMatcher = TOKEN_SCRIPT_PATTERN.matcher(tokenScript);
-                        if (tokenMatcher.find()) {
-                            foundToken = true;
-                        }
+            var elementList = new ArrayList<>(elements);
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java]: why do you create a copy here?

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java [hunk 8]
```diff
@@ -123,28 +119,29 @@ public void parseTokenData() throws IOException {
...
-        }
-        request = new HttpGet("https://music.apple.com" + tokenScriptURL);
-        try (var response = this.httpInterfaceManager.getInterface().execute(request)) {
-            var tokenScript = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8);
-            var tokenMatcher = TOKEN_SCRIPT_PATTERN.matcher(tokenScript);
-            if (!tokenMatcher.find()) {
-                throw new IllegalStateException("Cannot find token in script");
+
+            for (var element : elements) {
+                String tokenScriptURL = element.attr("src");
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java:130]:
```suggestion
                var tokenScriptURL = element.attr("src");
```

**Review by TopiSenpai — CHANGES_REQUESTED** [review]

File: main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java (hunk 8 above)
Comment (Jacrispys) [reply main/src/main/java/com/github/topisenpai/lavasrc/applemusic/AppleMusicSourceManager.java]: stupid var's lol

**Review by TopiSenpai — APPROVED** [review]

**Pull request merged**